doc/adr/0017-norm-test-files.md

```diff
@@ -0,0 +1,57 @@
+# 17. Test files for norms
+
+**Date**: 2025-03-20
+
+## Status
+
+Proposal
+
+## Context
+
+We currently have a lot of xml files and also xml-strings in use for backend unit and integration tests.
+
+The xml-strings in tests create some challenges when adjusting them for new LegalDocMl.de versions as it is harder to automatically update them.
+
+There are also some challenges with the xml files, mostly stemming from them being all thrown in a folder without much organization. This makes it hard to:
+* know which xml-file(s) to use if one needs "just a norm"
+* finding the correct related norms. E.g. the Zielnorm of a Änderungsgesetz
+* figuring out the eli of a norm, as it is often needed in other parts of tests
+* choosing multiple xml-file(s) for multiple norms as many xml-files use the same elis
+* creating a norm that includes Offenestrukturen or binary attachments (all need to be identified and loaded individually and then combined)
+
+Additionally, some tests files only include very minimal data that makes them schema-invalid, which creates challenges with using them in certain scenarios.
+
+**We can not use real norms imported from the migration but need to use our own test norms**
+
+## Decision
+
+1. Xml-Strings may only be used for short snippets. Whole norms should always be loaded from a xml-file.
+2. We organize the xml files in two categories of test-norms:
+   1. General test norms
+      1. can be used by any test, and should thus be "just" some norms that could happen to be in our system during normal use
+      2. are stored in a folder structure according to their manifestation elis
+      3. are schema valid
+      4. should be documented in a README.md file explaining what norm it is (e.g. "Änderungsgesetz for the Vereinsgesetz")
+   2. Test specific test norms
+      1. may only be used in tests for one class
+      2. stored in the `test-specific` subfolder
```

Comment:
You mean in the corresponding test package? Like for example adapter/input/restapi/controller package for all unit controller tests?